Add "Generate UUID" and "Analyse UUID" operations

bartvanandel opened the pull request:
This is an update to the (closed, but unmerged) #1531, now with additional support for UUID v6 and v7.

Fixes #1216

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/gchq/CyberChef?pullRequest=2011) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/gchq/CyberChef?pullRequest=2011) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: victorlpgazolli<br/>:x: bartvanandel<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/gchq/CyberChef?pullRequest=2011) it.</sub>

a3957273 commented:
Hey!  UUID is one of our most used operations, so commits to make it even better are much appreciated.  Is there any chance we can keep the default as UUIDv4?  A lot of people have preset scripts that go to, e.g:

    /#recipe=Generate_UUID()

And I don't to break backwards compatibility given how often this is used.

bartvanandel commented:
@a3957273 I've added `defaultIndex`, as well as a unit test to ensure it works correctly. Ran into an issue there however, see #2015.

I also simplified the argument names a bit by moving most of the info to the `hint` properties. Looks like those aren't rendered correctly for `option` and `string` fields at the moment though. I ran `$('[data-toggle=tooltip]').tooltip()` from the Developer Console, which does change the DOM, but no tooltips are being displayed in Safari or Chrome (both on MacOS). Looks like this should run automatically, which it doesn't either. I'll see if I can find out how to fix this, but that's going to be another PR for sure.

bartvanandel commented:
> Looks like those aren't rendered correctly for `option` and `string` fields at the moment though. I ran `$('[data-toggle=tooltip]').tooltip()` from the Developer Console, which does change the DOM, but no tooltips are being displayed in Safari or Chrome (both on MacOS). Looks like this should run automatically, which it doesn't either. I'll see if I can find out how to fix this, but that's going to be another PR for sure.

Investigated and fixed here: #2016

a3957273 commented:
Woo!  Thanks, this is really useful.  Also enjoy the changes to get defaultIndex working.  Thanks for the contribution.
